subsys/logging/log_backend_net.c

```diff
@@ -216,10 +216,32 @@ static void panic(struct log_backend const *const backend)
 	panic_mode = true;
 }
 
+static void sync_string(const struct log_backend *const backend,
+		     struct log_msg_ids src_level, u32_t timestamp,
+		     const char *fmt, va_list ap)
+{
+	u32_t flags = LOG_OUTPUT_FLAG_LEVEL | LOG_OUTPUT_FLAG_FORMAT_SYSLOG |
+		LOG_OUTPUT_FLAG_TIMESTAMP;
+	u32_t key;
+
+	if (!net_init_done) {
+		net_init_done = true;
+		if (do_net_init() < 0) {
+			net_init_done = false;
+		}
+	}
+
+	key = irq_lock();
+	log_output_string(&log_output, src_level, timestamp, fmt, ap, flags);
+	irq_unlock(key);
+}
+
 const struct log_backend_api log_backend_net_api = {
-	.put = send_output,
 	.panic = panic,
 	.init = init_net,
+	.put = IS_ENABLED(CONFIG_LOG_IMMEDIATE) ? NULL : send_output,
+	.put_sync_string = IS_ENABLED(CONFIG_LOG_IMMEDIATE) ?
```

Comment:
you haven't add support for synchronous hexdump printing.

Reply:
Yes, probably does not make sense to send hexdumps to remote syslog servers so I left it out.

Reply:
i see. Maybe it's work putting comment here about this decision. In future some may wonder.

Reply:
Sure, I will add comment about that.